Character.cpp

```diff
@@ -98,20 +106,44 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        initialStats(hitPoints, *initialHitPoints);
+        initialStats(armor, *initialArmorLevel);
+        initialStats(attackDamage, *initialAttackDamage);
+
+        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl; 
     }
 }
+ 
+ void Character::boostArmor( double amount )
+    {
+        armor += amount;
+        std::cout << getName() << "'s armor level has been boosted to " << armor << std::endl;
+    }
+    
+    void Character::boostHitPoints( double amount )
+    {
+        hitPoints += amount;
+        std::cout << getName() << "'s hit point level has been boosted to " << hitPoints << std::endl;
+    }
+    
+    void Character::boostAttackDamage( double amount )
+    {
+        attackDamage += amount;
+        std::cout << getName() << "'s attack damage level has been boosted to " << attackDamage << std::endl;
+    }
```

Comment:
None of these functions needed to be moved from the header file.
Don't change code that doesn't need to be changed.
- [ ] revise